### ISSUE TITLE
Automatically Get `play.api.libs.json.Reads` and `Writes` via `circe`?

### ISSUE THREAD

**Issue by kevinmeredith** [jilen/play-circe#4]
Given a case class with > 22 fields, i.e. https://github.com/playframework/playframework/issues/1141#issuecomment-82398165, as I understand, I need to manually build the `Reads[X]` and `Writes[X]` - http://stackoverflow.com/a/28168370/409976. I'd prefer to automatically get them, vis-a-vis [scala - parse json of more than 22 elements into case class](http://stackoverflow.com/questions/33318802/scala-parse-json-of-more-than-22-elements-into-case-class) in `circe`.

However, I'd prefer to keep using `play json` since my entire project uses it, rather than `circe`. Is it possible to do:

```
import play.api.libs.json.{Reads, Writes}

case class MoreThan22Fields(x1: String, x2: Int, ..., x40: Boolean)
def f(x: MoreThan22Fields): Reads[MoreThan22Fields] = ???
def g(x: MoreThan22Fields): Writes[MoreThan22Fields] = ???
```

where the `???` definition is replaced by the functions, respectively:

```
MoreThan22Fields => CirceJsonDecoder[MoreThan22Fields] => Reads[MoreThan22Fields]
MoreThan22Fields => CirceJsonEncoder[MoreThan22Fields] => Writes[MoreThan22Fields]
```

Note that my understanding is:

```
encode : Object -> JSON
decode : JSON  -> Object
```

In short, I'd like to use the power of `circe` per > 22 field case classes, but I'd like to use Play's `Reads` and `Writes`.

How can I do this?

**Comment by jilen** [comment]
Is this https://github.com/xdotai/play-json-extensions what you are looking for ?
